library/alloc/src/str.rs

```diff
@@ -400,13 +402,13 @@ impl str {
         }
         return s;
 
-        fn map_uppercase_sigma(from: &str, i: usize, to: &mut String) {
+        fn map_uppercase_sigma(from: &str, i: usize) -> char {
             // See https://www.unicode.org/versions/Unicode7.0.0/ch03.pdf#G33992
             // for the definition of `Final_Sigma`.
             debug_assert!('Σ'.len_utf8() == 2);
             let is_word_final = case_ignorable_then_cased(from[..i].chars().rev())
                 && !case_ignorable_then_cased(from[i + 2..].chars());
-            to.push_str(if is_word_final { "ς" } else { "σ" });
+            if is_word_final { 'ς' } else { 'σ' }
         }
```

Comment:
Actually, I think the old version is better here, as it avoid having to encode the `char` to UTF-8 and can simply do a memory store. Would you mind going back the the old version or returning a `&'static str` instead?

Reply:
I had a look to the assembly and actually the compiler produces a very optimized code tailored to the only 2 possible values returned by the function, it all boils down to a `or` and two `move` to memory
If you wish you can have a look here at a short example https://godbolt.org/z/3vrq6114q

Reply:
Yeah, you're right! In that case, this is fine (and perhaps even a bit neater).